src/main/java/core/basesyntax/Storage.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruits = new HashMap<>();
+
+    public static void add(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public static void subtract(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
```

Comment:
Ensure that the resulting quantity after subtraction is not negative. If it is, throw a `RuntimeException` to prevent invalid storage states. This aligns with the checklist requirement: 'Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.'